### PR TITLE
Fixed some issues found by clang-tidy-7

### PR DESCRIPTION
**Pull request by fweik**

### CODE REVIEW

File: src/core/cells.hpp [hunk 1]
```diff
@@ -72,15 +72,18 @@
...
-#define CELL_STRUCTURE_NONEYET -1
-/** Flag indicating that the current cell structure will be used further on */
-#define CELL_STRUCTURE_CURRENT 0
-/** cell structure domain decomposition */
-#define CELL_STRUCTURE_DOMDEC 1
-/** cell structure n square */
-#define CELL_STRUCTURE_NSQUARE 2
-/** cell structure layered */
-#define CELL_STRUCTURE_LAYERED 3
+enum {
```
Comment (jngrad) [inline src/core/cells.hpp:71]: Here the Doxygen syntax gets broken. The group description should become an enum description, e.g.:
```c++
/** Cell Structure */
enum {
  /** Flag indicating that there is no cell system yet. Only at the
   *  VERY beginning of the code startup.
   */
  CELL_STRUCTURE_NONEYET = -1,
  /** Flag indicating that the current cell structure will be used further on */
  CELL_STRUCTURE_CURRENT = 0,
  /** cell structure domain decomposition */
  CELL_STRUCTURE_DOMDEC = 1,
  /** cell structure n square */
  CELL_STRUCTURE_NSQUARE = 2,
  /** cell structure layered */
  CELL_STRUCTURE_LAYERED = 3
};
```

File: src/core/domain_decomposition.cpp [hunk 2]
```diff
@@ -71,28 +61,11 @@ double max_skin = 0.0;
 /*@{*/
 
 /** Convenient replace for loops over all cells. */
```
Comment (jngrad) [inline src/core/domain_decomposition.cpp]: please remove the orphaned Doxygen lines here, and at lines 65, 67-68

File: src/core/electrostatics_magnetostatics/p3m-dipolar.cpp [hunk 3]
```diff
@@ -1826,7 +1826,8 @@ static double dp3m_m_time(char **log, int mesh, int cao_min, int cao_max,
   /* the initial step sets a timing mark. If there is no valid r_cut, we can
      only try
      to increase cao to increase the obtainable precision of the far formula. */
-  do {
+  double tmp_time;
+  -do {
```
Comment (jngrad) [inline src/core/electrostatics_magnetostatics/p3m-dipolar.cpp]: extra minus sign

File: src/core/cells.hpp [hunk 4]
```diff
@@ -67,20 +67,22 @@
...
+  CELL_STRUCTURE_CURRENT = 0,
+  /** cell structure domain decomposition */
+  CELL_STRUCTURE_DOMDEC = 1,
+  /** cell structure n square */
+  CELL_STRUCTURE_NSQUARE = 2,
+  /** cell structure layered */
+  CELL_STRUCTURE_LAYERED = 3
+};
+
 /*@}*/
```
Comment (jngrad) [inline src/core/cells.hpp]: this line still needs to go, it is part of the old Doxygen group
```suggestion

```

File: src/core/MpiCallbacks.cpp [hunk 5]
```diff
@@ -76,7 +76,12 @@ int MpiCallbacks::add(func_ptr_type fp) {
 
 void MpiCallbacks::remove(const int id) { m_callbacks.remove(id); }
 
-void MpiCallbacks::abort_loop() const { call(LOOP_ABORT, 0, 0); }
+void MpiCallbacks::abort_loop() const {
+  try {
+    call(LOOP_ABORT, 0, 0);
+  } catch (...) {
```
Comment (KaiSzuttor) [inline src/core/MpiCallbacks.cpp:82]: why is that needed?

Comment (fweik) [reply src/core/MpiCallbacks.cpp:82]: Because it's used in the destructor, which is not allowed to throw, but mpi can throw. When there is an error in MPI during shut down there is nothing we can do, so catch all should be fine.

**Review by KaiSzuttor — APPROVED** [review]
